binarytree/__init__.py

```diff
@@ -314,6 +315,38 @@ def _get_tree_properties(root):
     }
 
 
+def _get_parent_node(root, node):
+    """Search from the binary tree and return the parent node for require node.
+
+    :param root: Root node of the binary tree.
+    :rtype: binarytree.Node
+    :param node: Require node you want to get its parent node.
+    :rtype: binarytree.Node
+    :return: The parent node of require node.
+    :rtype: binarytree.Node
+    """
+    root_height = root.height
+    node_height = node.height
+    if root_height == node_height:
+        return None
+    node_level_list = root.levels[root_height - node_height]
```

Comment:
Using `Node.levels` here is unnecessary and inefficient. You could search for parent using BFS instead.

Reply:
I will try this but I think BFS maybe also not the most efficient way. If you agree with this, I can add BFS as the first stage searching algorithm and later update it if I found a more efficient way.